### ISSUE TITLE
[Feature Request] Support for multiple clouds?

### ISSUE THREAD

**Issue by rustycl0ck** [openstack-exporter/openstack-exporter#139]
Right now there is an option to choose only a single cloud from the `clouds.yaml` file. Is there any plan to make it possible to scrape multiple openstack instances through the same exporter?

I know running multiple exporters might be a solution, but in my scenario, I need to scrape around 50 or more openstack instances, thus running so many exporters isn't really easily manageable.

**Comment by niedbalski** [comment]
I think is a good feature to consider, but I can't prioritize this for the next release. Is this something you are up to implement?

**Comment by rustycl0ck** [comment]
Sure @niedbalski, I can try to send a fix for this, but not sure how soon.

I glanced through the code and there are two ways to scrape multiple clouds:
1. To pass the list of clouds to scrape at the start time as cli parameter. This would mean that we always scrape all the clouds on every call by prometheus. This would require adding the custom label (maybe something like `cloud=test.cloud`) to all the metrics which are scattered over the code. Another issue with this approach is that if one of the clouds is either not reachable or wrong credentials, prometheus can't know that because it will just accept whatever metrics are provided by the exporter.
2. To have a list of all the available clouds in the `clouds.yaml` as usual, but instead of scraping all/multiple of them on every scrape, let the target come from prometheus (like `https://<exporter>:9180/probe?cloud=test.cloud`). With this approach we don't need to worry about changing any labels in the exporter code base (because the user can use `relabel_configs` in prometheus), and also, prometheus will get appropriate response (either 401 or 502 etc) if the required `cloud` can not be reached for some reason. Thus, prometheus will show it as target down, which can be very helpful for the user.

I'm looking to go with the second approach right now. But I think that might require some restructuring of the code. Let me know your thoughts.

Also just curious if there is any plan to move this project to the `prometheus-community` org? If so, I can also look at using the `promlog` library for logging (because I think the current amount of verbose/info logs might be an overkill if the exporter tries to scrape about 20-30 clouds per minute).

**Comment by rustycl0ck** [comment]
Fortunately I was able to spend time on this today and the functionality is working right now in the referenced PR, except that it breaks the `--collect-metric-time` feature (I'm guessing because the time collection metrics are not registered at the time of exporter creation but later in real time at the time of serving metric results).
I also took the liberty of copying the `Makefile.common` from one of the recent exporters (`prometheus-community/json_exporter`) because I was having a hard time with `grep` with the presence of the `vendor` directory.

Let me know if the current work is in the right direction. (Fixing the `--collect-metric-time` error might take some time)
